### ISSUE TITLE
Pasted UTF-8 characters into PowerShell shell are interpreted incorrectly.

### ISSUE THREAD

**Issue by RokeJulianLockhart** [PowerShell/PowerShell#24037]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Originally, this issue was going to be about how PowerShell can't handle certain legal UTF-8 characters in inode paths. However, after creating a test script:

```PS1
#!/usr/bin/env -S powershell

#```PS1
#Requires -PSEdition 'Core'
#Requires -Version 7
#```

Set-Location -LiteralPath "$HOME"

##### Example Directory
#Creates an example directory.

$DirectoryPath = 'TabPrefixed	ExampleDirectory.dir'
$PrintPath = "$DirectoryPath"
Write-Output (Show-Markdown -InputObject "The path is ``$PrintPath``.")

New-Item `
	-Path "$HOME" `
	-Name "$DirectoryPath" `
	-ItemType 'directory'

##### Example File
#Creates an example file.

$FilePath = 'TabPrefixed	ExampleFile.txt'
$PrintPath = "$FilePath"
Write-Output (Show-Markdown -InputObject "The path is ``$PrintPath``.")

New-Item `
	-Path ("$HOME" + [IO.Path]::DirectorySeparatorChar + "$DirectoryPath") `
	-Name "$FilePath" `
	-ItemType 'file' `
	-Value 'Example text.'

##### Path Acquisition
#Lists both.

$FullFilePath = (`
Get-ChildItem `
	-LiteralPath (
		"$HOME"          + [IO.Path]::DirectorySeparatorChar +
		"$DirectoryPath" + [IO.Path]::DirectorySeparatorChar +
		"$FilePath")
	)[0] | Select-Object -ExpandProperty 'FullName'

##### File Content Acquisition
#This should fail.

$PrintPath = "$FullFilePath"
Write-Output (Show-Markdown -InputObject "The path is ``$PrintPath``.")

Get-Content -LiteralPath "$FullFilePath"
Write-Output "$?"
```

I realized that it could. Instead, I've narrowed it to a fault in how PowerShell, via Konsole and VSCode's terminals, interprets certain characters (like `	` - tabs) as control codes:

```log
Get-Content -LiteralPath "/home/RokeJulianLockhart/Documents/pyside6/prefix^Iexample.txt/example.txt"
```

or appears to paste the path correctly, but doesn't recongise the value unless passed in a script:

```log
PS /home/RokeJulianLockhart> (Get-ChildItem -LiteralPath "$DirectoryPath")[0].FullName
/home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt
PS /home/RokeJulianLockhart> Get-Content -LiteralPath '/home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt'                                                                                                                                 
Get-Content: Cannot find path '/home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt' because it does not exist.
PS /home/RokeJulianLockhart> Get-Content -LiteralPath "$FullFilePath"                                                                                  
Example text.
PS /home/RokeJulianLockhart> Write-Output "$FullFilePath"            
/home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt
```

I've put the script section first, instead of the description of the issue, because my examples aforementioned utilize values from this script to demonstrate the problem.

### Expected behavior

```console
PS /home/RokeJulianLockhart> . '/home/RokeJulianLockhart/Documents/testpath/Test-Path.PS1'
The path is TabPrefixed ExampleDirectory.dir.



    Directory: /home/RokeJulianLockhart

UnixMode         User Group         LastWriteTime         Size Name
--------         ---- -----         -------------         ---- ----
drwxr-xr-x RokeJulian RokeJulian 10/07/2024 12:21            0 TabPrefixed      ExampleDirectory.dir
             Lockhart Lockhart
The path is TabPrefixed ExampleFile.txt.



    Directory: /home/RokeJulianLockhart/TabPrefixed     ExampleDirectory.dir

UnixMode         User Group         LastWriteTime         Size Name
--------         ---- -----         -------------         ---- ----
-rw-r--r-- RokeJulian RokeJulian 10/07/2024 12:21           13 TabPrefixed      ExampleFile.txt
             Lockhart Lockhart
The path is /home/RokeJulianLockhart/TabPrefixed        ExampleDirectory.dir/TabPrefixed        ExampleFile.txt.


Example text.
True
```

### Actual behavior

```console
PS /home/RokeJulianLockhart> (Get-ChildItem -LiteralPath "$DirectoryPath")[0].FullName
/home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt
PS /home/RokeJulianLockhart> Get-Content -LiteralPath '/home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt'                                                                                                                                 
Get-Content: Cannot find path '/home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt' because it does not exist.
PS /home/RokeJulianLockhart> Get-Content -LiteralPath "$FullFilePath"                                                                                  
Example text.
PS /home/RokeJulianLockhart> Write-Output "$FullFilePath"            
/home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt
```

### Error details

```console
Exception             : 
    Type                 : System.Management.Automation.ItemNotFoundException
    ErrorRecord          : 
        Exception             : 
            Type    : System.Management.Automation.ParentContainsErrorRecordException
            Message : Cannot find path '/home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt' because it does not exist.
            HResult : -2146233087
        TargetObject          : /home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt
        CategoryInfo          : ObjectNotFound: (/home/RokeJulianLoc…    ExampleFile.txt:String) [], ParentContainsErrorRecordException
        FullyQualifiedErrorId : PathNotFound
    ItemName             : /home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt
    SessionStateCategory : Drive
    TargetSite           : 
        Name          : ResolveDriveQualifiedPath
        DeclaringType : [System.Management.Automation.LocationGlobber]
        MemberType    : Method
        Module        : System.Management.Automation.dll
    Message              : Cannot find path '/home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt' because it does not exist.
    Source               : System.Management.Automation
    HResult              : -2146233087
    StackTrace           : 
   at System.Management.Automation.LocationGlobber.ResolveDriveQualifiedPath(String path, CmdletProviderContext context, Boolean allowNonexistingPaths, CmdletProvider& providerInstance)
   at System.Management.Automation.LocationGlobber.GetGlobbedMonadPathsFromMonadPath(String path, Boolean allowNonexistingPaths, CmdletProviderContext context, CmdletProvider& providerInstance)
   at System.Management.Automation.PathIntrinsics.GetResolvedPSPathFromPSPath(String path, CmdletProviderContext context)
   at Microsoft.PowerShell.Commands.ContentCommandBase.ResolvePaths(String[] pathsToResolve, Boolean allowNonexistingPaths, Boolean allowEmptyResult, CmdletProviderContext currentCommandContext)
TargetObject          : /home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt
CategoryInfo          : ObjectNotFound: (/home/RokeJulianLoc…    ExampleFile.txt:String) [Get-Content], ItemNotFoundException
FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetContentCommand
InvocationInfo        : 
    MyCommand        : Get-Content
    ScriptLineNumber : 1
    OffsetInLine     : 1
    HistoryId        : 12
    Line             : Get-Content -LiteralPath '/home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt'
    Statement        : Get-Content -LiteralPath '/home/RokeJulianLockhart/TabPrefixed    ExampleDirectory.dir/TabPrefixed        ExampleFile.txt'
    PositionMessage  : At line:1 char:1
                       + Get-Content -LiteralPath '/home/RokeJulianLockhart/TabPrefixed    Exa …
                       + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    InvocationName   : Get-Content
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : 
      0
      1
```

### Environment data

```powershell
{
  "OS": "Fedora Linux 40 (KDE Plasma)",
  "SerializationVersion": {
    "Major": 1,
    "Minor": 1,
    "Build": 0,
    "Revision": 1,
    "MajorRevision": 0,
    "MinorRevision": 1
  },
  "PSCompatibleVersions": [
    {
      "Major": 1,
      "Minor": 0,
      "Build": -1,
      "Revision": -1,
      "MajorRevision": -1,
      "MinorRevision": -1
    },
    {
      "Major": 2,
      "Minor": 0,
      "Build": -1,
      "Revision": -1,
      "MajorRevision": -1,
      "MinorRevision": -1
    },
    {
      "Major": 3,
      "Minor": 0,
      "Build": -1,
      "Revision": -1,
      "MajorRevision": -1,
      "MinorRevision": -1
    },
    {
      "Major": 4,
      "Minor": 0,
      "Build": -1,
      "Revision": -1,
      "MajorRevision": -1,
      "MinorRevision": -1
    },
    {
      "Major": 5,
      "Minor": 0,
      "Build": -1,
      "Revision": -1,
      "MajorRevision": -1,
      "MinorRevision": -1
    },
    {
      "Major": 5,
      "Minor": 1,
      "Build": -1,
      "Revision": -1,
      "MajorRevision": -1,
      "MinorRevision": -1
    },
    {
      "Major": 6,
      "Minor": 0,
      "Build": -1,
      "Revision": -1,
      "MajorRevision": -1,
      "MinorRevision": -1
    },
    {
      "Major": 7,
      "Minor": 0,
      "Build": -1,
      "Revision": -1,
      "MajorRevision": -1,
      "MinorRevision": -1
    }
  ],
  "PSRemotingProtocolVersion": {
    "Major": 2,
    "Minor": 3,
    "Build": -1,
    "Revision": -1,
    "MajorRevision": -1,
    "MinorRevision": -1
  },
  "Platform": "Unix",
  "WSManStackVersion": {
    "Major": 3,
    "Minor": 0,
    "Build": -1,
    "Revision": -1,
    "MajorRevision": -1,
    "MinorRevision": -1
  },
  "GitCommitId": "7.4.3",
  "PSEdition": "Core",
  "PSVersion": {
    "Major": 7,
    "Minor": 4,
    "Patch": 3,
    "PreReleaseLabel": null,
    "BuildLabel": null
  }
}
```

**Comment by 237dmitry** [comment]
How it could be represented? As in bash?

![ArcoLinux_2024-07-10_18-26-58](https://github.com/PowerShell/PowerShell/assets/78153320/ab9f7bf9-d0c0-498f-a197-5ebcdd0420f4)

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/24037#issuecomment-2220837124

@237dmitry, I've asked about that at [SE SuperUser](https://superuser.com/questions/1848192/why-are-utf-8-paths-differently-escaped-in-outputs-across-core-cli-utilities#comment2926151_1848195), and the answer is such that it does appear feasible. Especially, for PowerShell, these utilities don't even need to be able to detect other shells' syntaxes (like `ls` does). Certainly, I can't think of any other solution this, else I would have filed this as an FR (rather than the BR it is).

**Comment by 237dmitry** [comment]
> should actually accept verbatim the characters

Yes, I think so. Maybe with ``` `t ``` or `\t` replacement.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/24037#issuecomment-2220984801

@237dmitry, my point is merely that the user shouldn't need to manually modify the path before pasting it.

**Comment by kasini3000** [comment]
1 Usually, saving the script as UTF8 encoding with BOM header can solve the problem of file names with special encoding in the script. But this is not compatible with “#!"
2 I usually use mkdir - p 'directory name with utf8 encoding'

**Comment by rhubarb-geek-nz** [comment]
Are you saying that pasting into a session where PowerShell is running should be context sensitive in how it interprets tab characters? 

For instance if you paste a tab as part of a filename, how is PowerShell to know you want that as part of the file name rather than tab-expansion to cycle through available files?

If you want this to be context sensitive then is this part of [PReadLine](https://github.com/PowerShell/PSReadLine) rather than PowerShell?

**Comment by mklement0** [comment]
For context:
* The concept of a character-encoding such as UTF-8 doesn't apply to in-memory strings getting pasted.
  * Scripts and files without a BOM default to UTF-8 in PowerShell Core; the only reason to use a BOM is for cross-edition scripts that Windows PowerShell too must interpret correctly. 
* Technically, a tab character is in the ASCII subrange of the Unicode "alphabet", and is a so-called [C0 control character](https://en.wikipedia.org/wiki/C0_and_C1_control_codes#C0_controls).

* Pasting into a terminal in most environments is done by _simulated typing_, unfortunately. In such environments pasting something with a tab causes that tab char. to be interpreted as a request for tab-completion, thereby effectively discarding the tab while possibly altering what is pasted.
  * Overcoming this limitation is discussed in the following PSReadLine issues:
    * https://github.com/PowerShell/PSReadLine/issues/579
    * For Unix: https://github.com/PowerShell/PSReadLine/issues/1471

* While the old-style `conhost.exe` console windows on Windows do support direct pasting, they seemingly _discard_ tab characters.

* macOS terminals either quietly convert a pasted tab to a _space_ (Terminal.app) or prompt for whether the tabs should be pasted (iTerm.app), but the pasting then again happens via simulated typing.

I'm unclear on whether PSReadLine even has the option to override these behaviors.

---

As for _tab-completing_ a filename with embedded tabs (not supported on Windows), as observed on macOS:
* `bash` completes with a _literal_ tab char., properly escaped
* PowerShell completes with the tab char. too, but it is _rendered as_ `^I` and _not escaped_, which could be considered a bug (to make it work, the `^I` must either be `` ` ``-escaped, or the whole argument must be quoted).

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/24037#issuecomment-2223724193

@mklement0, that's a brilliantly comprehensive explanation. Would you consider this issue a duplicate of https://github.com/PowerShell/PSReadLine/issues/1471#issue-600646593, or should I file a separate issue at that repository?

**Comment by mklement0** [comment]
Thanks, @RokeJulianLockhart.

https://github.com/PowerShell/PSReadLine/issues/1471#issue-600646593 may be a duplicate - I'm not certain if tabs would be retained as-is as part of a bracketed past.

I see that you opened a new PSReadLine issue: I suggest updating it with the above information.

It looks like the Windows Terminal PR you linked to was merged quite a while ago, so I assume the feature is already in place.

**Comment by 237dmitry** [comment]
> The concept of a character-encoding such as UTF-8 doesn't apply to in-memory strings getting pasted

When using `Get-Clipboard`, tab characters are not replaced with spaces.:

```
$ [int[]][char[]] "a    b"    # pasted from clipboard with Ctrl+V
97
32
32
32
32
98

$ [int[]][char[]]  (Get-Clipboard)
97
9
98
```

**Comment by daxian-dbw** [comment]
@RokeJulianLockhart As shown in the screenshot below, you copied the text from the terminal output before pasting to `Get-Content -LiteralPath`. I believe when you copy the terminal output, you already lost the tab characters `\t`. Can you please double check if that's the case? You can check by pasting to VSCode editor after copying from the terminal output, and see if the tab has been replaced with spaces already.

![image](https://github.com/user-attachments/assets/ae10608d-0e79-425c-9f7b-759da19903ef)

**Comment by mklement0** [comment]
@daxian-dbw, while it is true that tab chars. are lost when copying output from VSCode's integrated terminal (they become spaces), the problem exists even if you paste a string with bona fide tab characters.

**Comment by daxian-dbw** [comment]
Yes, I understand. Tab is intentionally replaced with 4 space characters (see [the code](https://github.com/PowerShell/PSReadLine/blob/eff1452a647a0bb5863cf5fa42062b3b1f701c10/PSReadLine/KillYank.cs#L675-L676)), and it was a by design change: https://github.com/PowerShell/PSReadLine/issues/144.

I just want to point out that the scenario described by this issue -- copy the terminal output and paste in your command -- won't work even if PSReadLine doesn't alter the Tab character. And I don't think it's worth breaking the current behavior.

**Comment by mklement0** [comment]
@daxian-dbw: 

Good to know about PSReadLine's tabs-to-spaces conversion.

> copy the terminal output and paste in your command -- won't work even if PSReadLine doesn't alter the Tab character.

That may be true for VSCode, but it isn't true for other terminal emulators, such as on macOS - on copying, they _do_ preserve literal tab chars. in the selection.

---

Let me try to summarize:

* PSReadLine converts each literal tab char. pasted into 4 spaces by design; Bash's readline library appears to do a similar thing (1 space). Additionally, terminal applications such as iTerm.app on macOS themselves offer modifying the pasted text if it contains tab chars., before passing it to the target shell.

  *  For pasting _code_ this makes sense, but it prevents pasting file paths that contain embedded tab chars. - though that is definitely an edge case.

* PowerShell's tab-completion completes file names with embedded tab chars. as such, but renders them as `^I` (which is just a _display_ problem). However, it neglects to _quote_ such file names, which is arguably a bug (but, again, an edge case).

**Comment by AngelRazo68** [comment]
bueno señores. solo quiero comentarles, yo en realidad solo es que no së, todo esto es nuevo para mi, nunca fui a la escuela, nunca tuve un profesor que me enseñara, nunca estudie mi maximo grado de estudios es la secundaria concluida. eso es todo que en estos ultimos tiempor he hecho muchas cosas desde armar esta chatarra del basurero, pero para cualquiera de ustedes hubiera sido una tarea sencilla, para mi fue una tormenta en la que me meti. pues no sabia ni como ni por donde. ni nada, ni terminos cientificos ni terminos de ninguno. solo hago las cosas, estudio lo que veo, practico lo que puedo, pero no culpo a nadie si soy yo, y saben mucho de lo que hago es solo causarle problemas a microsoft. por que soy necio. pero aprendo, y creo que aprendo bien, si bien no eh hecho tal vez para todos ustedes algo bien, pues es normal, no estudie nada. pero a mi punto de vista, yo se mucho mas que muchos que estudian, yo aprendo solo, con tutoriales, con practica aun que luego me pierdo, pero como sea regreso.  y todo esto lo aprendo por gusto. por que me gusta, por que quiero aprender, por que ademas lo necesito, necesito generar dinero sin necesidad de tener que acudir a una empresa negrera que solo ocupa a la gente por horas y horas y nunca se llenan y como consegui esos empleos, pues para mi fue simple, examenes y examenes, y examenes, y creo que hasta suerte tenia, por que donde me pare me quede. sin saber absolutamente nada, pero no soy de las personas que se queda esperando, si observo, y aprendo. y jamas me corrieron por malo. al contrario. tengo muchos reconocimientos  de donde trabaje, en cada uno de mis trabajos siempre sobresali de alguna manera, perjo jamas me dijeron que yo fuera malo. al contrario, siermpre en poco tiempo siempre me mantenia en los primeros lugares de las empresas. y de igual manera si me sali de todas fue por alguna razon, y tambien se que a la que quiera regresar puedo hacerlo.  pero una no kiero, y dos no puedo. vivo con una persona de mas de 80 años que por cierto, yo no tube hijos, y no se lo que es cuidar niños chiquito. dicen que es lo mismo. y honestamente yo no hubiera sido buen padre, no tengo la pasciencia de poder tolerar tantas grocerias. y a mi madre le aguanto mucho. pero hasta el dia de hoy con todo esto que estoy medio aprendiendo y con mi madre, voy a terminar loco.  es pero que no. pero si lo eh pensado. señores, yo no soy para estar aqui, solo se que voy a seguir haciendo barbaridades con sus sistemas, y  al final almenos yo me doy cuenta que ahorita para mi es muy sencillo hacer muchas cosas que antes no tenia ninguna idea de para que diablos eran sus codificaciones,  solo aqui estoy aqui voy a seguir, tal vez no en lo que ustedes pensaban, pero voy a crear, voy a generar paginas, aplicaicones, y no me importa si es java. si es phyton, no mie importa el programa, ami lo que me importa es llegar a la meta, y la meta es llegar con la gente y que le guste lo que yo hago. solo eso.  y perdon, si la eh cajeteado de mas, pero aveces me desesperan tantos candados. yo seria mas practico. y solo por no haber estudiado. pero con la edad que yo tengo y si hubiera estudiado,, que creen que todos ustedes los podria haber tenido babeando de lo que podria hacer. pero no estarde. y me quedan unos cuantos años para hacer algo. asi que les agradezco sus criticas que poco entendi. pero al final entendi algo por que si me daba cuenta de lo que decian solo que con mis terminos, osea mas burocrata. pero al final , gracias. por que gracias a ustedes existe esta y muchas paginas para burros como yo. y perdon los tengo que dejar, me tengo que poner a generar algo.  me urge dinero, por que como dicen esta panza no es de salario minimo. 
cualquier cosa aqui me tendran, que yo jamas bloqueo a nadie que no me hace nada. y menos a ustedes que lo unico que quicieran, es que yo de verdad supiera lo que tal vez pensaban que se.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/24037#issuecomment-2227127337

@AngelRazo68, what...?

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/24037#issuecomment-2224029814

@daxian-dbw, my sincere apologies for the wait to respond. I've been spending some time diagnosing some related problems, like https://github.com/microsoft/debugpy/issues/1671#issuecomment-2379927310, and then migrating wholesale away from tabs in file names. [^1]

Irrespective, it certainly does appear to be replaced, and solely by `pwsh` - not `bash`:

> ![Screenshot_20241022_175005](https://github.com/user-attachments/assets/0de43f63-a471-4349-8c82-53df23fa3e02)

[^1]: https://stackoverflow.com/revisions/79048760/10#:~:text=key%20per%20line.-,Solution,-The%20sole%20readable%2C

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/24037#issuecomment-2224000265

@237dmitry, I'd try `Get-Clipboard`, but I'm blocked by https://github.com/Pscx/Pscx/issues/103#issue-2601956172. Consequently, are you using Windows? If so, I'm surprised you can use tabs in filenames, considering that they're explicitly illegal in filenames created via the Win32 API (as all early UTF-8/ASCII whitespace characters are).

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/24037#issuecomment-2223850547

@mklement0, thanks for the advice, especially per https://github.com/PowerShell/PSReadLine/issues/4072#issuecomment-2224037550. Do you suggest that I merely duplicate the content of https://github.com/PowerShell/PowerShell/issues/24037#issuecomment-2223724193 into the technical implementation section?

**Comment by mklement0** [comment]
I suggest opening a new PSReadLine issue that succinctly describes the problem; feel free to adapt / copy the text from my comment in the process.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/24037#issuecomment-2429893422

@mklement0, this is a lot to get my head around. Summarily, should I request that PSReadLine not escape tab characters unless at the start of a line?

**Comment by mklement0** [comment]
@RokeJulianLockhart, you could ask that PSReadLine paste tabs as-is, though given [@daxian-dbw's comment above](https://github.com/PowerShell/PowerShell/issues/24037#issuecomment-2225934625), I'm not sure such a change will be considered.

If it were to be considered, note that https://github.com/PowerShell/PSReadLine/issues/1471 must be implemented first, because, without paste bracketing, pasting involves _simulated typing_, so that a tab char. as part of a pasted string would trigger tab-completion
(this is what's currently happening on macOS and Linux).

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/24037#issuecomment-2634587883

@mklement0, thanks. Filed at https://github.com/PowerShell/PSReadLine/issues/4447#issue-2830989421.
